packages/sdk/test/integration/quantum-policy.test.ts

```diff
@@ -0,0 +1,170 @@
+
+        publisherIdentity = MLDSAKeyPairIdentity.generate()
+        subscriberIdentity = MLDSAKeyPairIdentity.generate()
+
+        // Client for setting up the stream
+        nonQuantumClient = environment.createClient({
+            auth: {
+                identity: EthereumKeyPairIdentity.generate(),
+            }
+        })
+        // nonQuantumClient also has pub/sub permissions to stream because it created the stream
+        stream = await createTestStream(nonQuantumClient, module)
+        await stream.grantPermissions({ permissions: [StreamPermission.PUBLISH], userId: await publisherIdentity.getUserIdString() })
+        await stream.grantPermissions({ permissions: [StreamPermission.SUBSCRIBE], userId: await subscriberIdentity.getUserIdString() })
+    })
+
+    afterEach(async () => {
+        await environment.destroy()
+    })
+
+    describe('pubsub under strict quantum settings', () => {
+
+        let sub: Subscription
+
+        beforeEach(async () => {
+            // Clients for pub/sub (can't make transactions)
+            quantumPublisher = environment.createClient({
+                auth: {
+                    identity: publisherIdentity,
+                },
+                encryption: {
+                    requireQuantumResistantSignatures: true,
+                    requireQuantumResistantKeyExchange: true,
+                    requireQuantumResistantEncryption: true,
+                }
+            })
+            quantumSubscriber = environment.createClient({
+                auth: {
+                    identity: subscriberIdentity,
+                },
+                encryption: {
+                    requireQuantumResistantSignatures: true,
+                    requireQuantumResistantKeyExchange: true,
+                    requireQuantumResistantEncryption: true,
+                }
+            })
+
+            sub = await quantumSubscriber.subscribe({
+                streamId: stream.id,
+            })
+        })
+
+        it('works between quantum identities', async () => {
+            const sub = await quantumSubscriber.subscribe({
+                streamId: stream.id,
+            })
+            const testMsg = Msg()
+            await quantumPublisher.publish(stream.id, testMsg)
+            const received = []
+            for await (const msg of sub) {
+                received.push(msg.content)
+                if (received.length === 1) {
+                    break
+                }
+            }
+            expect(received).toEqual([testMsg])
+        })
+
+        // Need to use done callback instead of async/await because we use the error listener
+        it('fails if requirements are violated', (done) => {
+            const testMsg = Msg()
+
+            sub.on('error', (err: Error) => {
+                expect(err.message).toContain('signature')
+                done()
+            })
+
+            nonQuantumClient.publish(stream.id, testMsg)
```

Comment:
If the test can have assertions at the and of the test it is maybe bit more readable. In this case it doesn't matter too much, but we've previously refactored some larger tests to use that kind of style:

```suggestion
            const errorPromise = new Promise<Error>((resolve) => {
                sub.on('error', resolve)
            })
            nonQuantumClient.publish(stream.id, testMsg)
            const err = await errorPromise
            expect(err.message).toContain('signature')
```